### PR TITLE
Add Drop page for minting NFTs with partners

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on enhancing the `DashboardLayout` and `DropPage` components by adding error handling, metadata generation, and support for new blockchain features. It also includes updates to the `NftMint` component for improved user interaction and visual representation.

### Detailed summary
- Added `AppFooter` and `ErrorProvider` to `DashboardLayout`.
- Introduced `DropPageData` type in `data.ts`.
- Updated `DROP_PAGES` with new NFT metadata and thumbnails.
- Enhanced `DropPage` for dynamic metadata generation.
- Improved `NftMint` component with quantity selection and custom address minting.
- Integrated `CustomConnectWallet` for wallet connection handling.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
* **#5307** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5307?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/public/drop/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Medias in this folder are used for /drop/<slug>
```
Comment (MananTank) [inline apps/dashboard/public/drop/README.md:1]: Remove this folder, move the media in app router folder next to the page and use `import ...` syntax for importing the image

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  slug: string;
+  contractAddress: string;
+  chainId: number;
+  hideQuantitySelector?: boolean;
+  hideMintToCustomAddress?: boolean;
+  // If not defined, we will use the image of the NFT or contract's image
+  thumbnail?: string;
+} & ({ type: "erc1155"; tokenId: bigint } | { type: "erc721" });
+
+export const DROP_PAGES: DropPageData[] = [
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:27]: Remove export here - exporting non-next things from page.tsx is not valid

Comment (kien-ngo) [reply apps/dashboard/src/app/drops/[slug]/page.tsx:27]: no way I didn't know this!

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      displayName={displayName || ""}
+      thumbnail={thumbnail}
+      description={description || ""}
+      currencySymbol={currencySymbol}
+      pricePerToken={pricePerToken}
+      noActiveClaimCondition={false}
+      {...project}
+    />
+  );
+}
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:134]: <i class='graphite__hidden'>[Re: line +135]<br/><br/></i>

Add page metadata - title, description, opengraph etc<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5307?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        contract: getContract({
+          address: claimCondition.currency,
+          chain,
+          client,
+        }),
+      })
+    : undefined;
+
+  const currencySymbol = currencyMetadata?.symbol || "";
+  if (!currencyMetadata?.decimals) {
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:115]: use instead:
```
if (!currencyMetadata) {
    return notFound();
  }
```

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+  const displayName = contractMetadata.name || nft.metadata.name || "";
+
+  const description =
+    contractMetadata.description || nft.metadata.description || "";
+
+  if (!claimCondition) {
+    return (
+      <NftMint
+        contract={contract}
+        displayName={displayName || ""}
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:95]: type of `displayName` is already `string` - no fallback to empty string required

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    getContractMetadata({ contract }),
+  ]);
+
+  const thumbnail =
+    project.thumbnail || nft.metadata.image || contractMetadata.image || "";
+
+  const displayName = contractMetadata.name || nft.metadata.name || "";
+
+  const description =
+    contractMetadata.description || nft.metadata.description || "";
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:89]: `contractMetadata.description` is typed as `any` and so `description` also ends up with `any` type - add a `typeof` check to `contractMetadata.description` to fix this

**Review by MananTank — CHANGES_REQUESTED** [review]
please address comments - mostly nits

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    chainId: 543210,
+    hideMintToCustomAddress: true,
+    hideQuantitySelector: true,
+    thumbnail: "/drop/thumbnails/zero-x-thirdweb.mp4",
+  },
+];
+
+export default async function DropPage({
+  params,
+}: { params: { slug: string } }) {
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:52]: params should be a promise

```
Promise<{ slug: string }>
```

File: apps/dashboard/src/app/drops/[slug]/page.tsx (hunk 5 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/drops/[slug]/page.tsx]: updated

File: apps/dashboard/src/app/drops/[slug]/page.tsx (hunk 4 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/drops/[slug]/page.tsx]: updated

File: apps/dashboard/src/app/drops/[slug]/page.tsx (hunk 7 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/drops/[slug]/page.tsx]: updated

File: apps/dashboard/src/app/drops/[slug]/page.tsx (hunk 6 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/drops/[slug]/page.tsx]: updated

File: apps/dashboard/src/app/drops/[slug]/page.tsx (hunk 3 above)
Comment (kien-ngo) [reply apps/dashboard/src/app/drops/[slug]/page.tsx:154]: updated

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+      title: "ZERO x thirdweb",
+      description:
+        "This exclusive commemorative NFT marks the official launch of ZERϴ's mainnet and our exciting partnership with thirdweb. Own a piece of this milestone in blockchain history as we make onchain transactions free with zero.network",
+    },
+  },
+
+  // Add more chains here
+];
+
+export async function generateMetadata({
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:59]: This is wrong - add the return type `Promise<Metadata>` on `generateMetadata` to see the issue

Comment (MananTank) [reply apps/dashboard/src/app/drops/[slug]/page.tsx:59]: Also - make sure `title` and `description` inside `openGraph` object is also set in metadata

Comment (MananTank) [reply apps/dashboard/src/app/drops/[slug]/page.tsx:59]: use `notFound()` if you can't find the `project`

File: apps/dashboard/src/app/drops/[slug]/page.tsx [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+  if (!currencyMetadata) {
+    return notFound();
+  }
+
+  const pricePerToken = Number(
+    toTokens(claimCondition.pricePerToken, currencyMetadata.decimals),
+  );
+
+  return (
```
Comment (MananTank) [inline apps/dashboard/src/app/drops/[slug]/page.tsx:154]: Add opengraph-image.tsx file and dynamically generate OG image for this page

**Review by MananTank — COMMENTED** [review]
The current opengraph-image.tsx is not very useful - @kien-ngo leaving it upto you to update it later on. Approving the PR for now

**Pull request merged**